wps_tools/output_handling.py

```diff
@@ -0,0 +1,181 @@
+    """
+    Access content from a rda url file as a Rpy2 vector object
+
+    Parameters:
+        url (str): file or http url path to a rda file
+        vector_name (str): the name the vector was given when it
+            was saved to the rda file
+
+    Returns:
+        Rpy2 object: Rpy2 representation of the R object "vector_name"
+    """
+
+    with NamedTemporaryFile(
+        suffix=".rda", prefix="tmp_copy", dir="/tmp", delete=True, mode="wb"
+    ) as r_file:
+        urlretrieve(url, r_file.name)
+        return load_rdata_to_python(r_file.name, vector_name)
+
+
+def vector_to_dict(url, vector_name):
+    """
+    Access content from a rda url file as a Python dictionary
+
+    Parameters:
+        url (str): file or http url path to a rda file containing
+            a named vector object
+        vector_name (str): the name the vector was given when it
+            was saved to the rda file
+
+    Returns:
+        dictionary: Python dictionary representation of a named
+            R vector
+    """
+
+    with NamedTemporaryFile(
+        suffix=".rda", prefix="tmp_copy", dir="/tmp", delete=True, mode="wb"
+    ) as r_file:
+        urlretrieve(url, r_file.name)
+        vector = load_rdata_to_python(r_file.name, vector_name)
+
+    if "robjects.vectors.FloatVector" not in str(type(vector)):
+        raise ValueError(f"{vector_name} is not a vector")
+
+    else:
+        base = get_package("base")
+
+        return {
+            (base.names(vector)[index]): (
+                None if math.isnan(vector[index]) else vector[index]
+            )
+            for index in range(len(vector))
+        }
+
+
+def txt_to_string(url):
+    """
+    Access content from a txt url file as a string
+
+    Parameters:
+        url (str): file or http url path to a txt file
+
+    Returns:
+        string: content of the input txt file
+    """
+
+    with urlopen(url) as text:
+        return text.read().decode("utf-8")
+
+
+def get_available_robjects(url):
+    """
+    Get a list of all the objects stored in an rda file
+
+    Parameters:
+        url (str): file or http url path to a rda file
+
+    Returns:
+        list: a list of the names of objects stored in an rda file
+    """
+
+    with NamedTemporaryFile(
+        suffix=".rda", prefix="tmp_copy", dir="/tmp", delete=True, mode="wb"
+    ) as r_file:
+        urlretrieve(url, r_file.name)
+        return list(robjects.r(f"load(file='{r_file.name}')"))
+
+
+def auto_construct_outputs(outputs):
+    """
+    Automatically construct Python objects from input url files.
+    Written to construct complex WPS process Outputs.
+
+    Parameters:
+        outputs (list): list of file or http url paths to files
+
+    Returns:
+        list: the constructed python objects in a list
+    """
+
+    process_outputs = []
+    for value in outputs:
+        if value.endswith(".rda") or value.endswith(".rdata"):
+            vector_name = get_available_robjects(value)[0]
```

Comment:
So `get_available_robjects` returns a list, but we only ever take the first item from that list. Is it possible for there to be multiple items?

Reply:
This is just the auto function, so a user can still manually gather other options, that being said I think it would be pretty easy to iterate through them. I wrote this before the break and kind of forgot about it.